terminal/src/Format.hs

```diff
@@ -133,6 +135,48 @@ format flags (Env inputs) =
     Project projectType paths ->
       formatFilesOnDisk flags projectType paths
 
+validate :: Env -> Task.Task Exit.Format ()
+validate (Env inputs) = do
+  case inputs of
+    Stdin ->
+      do
+        original <- Task.io BS.getContents
+        case formatByteString Parse.Application original of
+          Nothing ->
+            error "TODO: report error"
+          Just formatted ->
+            when (BSL.fromStrict original /= B.toLazyByteString formatted) $
+              Task.throw Exit.FormatValidateNotCorrectlyFormatted
+    Files paths ->
+      validateFiles paths
+    Project _ paths ->
+      validateFiles paths
+
+validateFiles :: [FilePath] -> Task.Task Exit.Format ()
+validateFiles paths = do
+  validationResults <- mapM validateFile paths
+  when (any (== False) validationResults) $
+    Task.throw Exit.FormatValidateNotCorrectlyFormatted
+
+validateFile :: FilePath -> Task.Task Exit.Format Bool
+validateFile path =
+  assertFileExists path >> Task.io (validateExistingFile path)
+
+validateExistingFile :: FilePath -> IO Bool
+validateExistingFile path = do
+  putStr ("Validating " ++ path)
+  original <- File.readUtf8 path
+  case formatByteString Parse.Application original of
```

Comment:
Similar to how `formatFile` does it, when `inputs` is `Project projectType paths` (line 152 above), we should use that projectType instead of `Parse.Application`.  That allows special core modules to be parsed properly.  (If you try it on the [gren/core](https://github.com/gren-lang/core) library, you should see the problem.)